### ISSUE TITLE
Compilation error with Pythia 8303

### ISSUE THREAD

**Issue by vvolkl** [HEP-FCC/dual-readout#10]
```
            cc:6:
     234    /tmp/vavolkl/spack-stage/spack-stage-dual-readout-0.0.2-fgymttq7gpq
            wpvxjqlgdjvplqotqdxw5/spack-src/Gen/include/Pythia8ToHepMC3.h: In m
            ember function 'bool HepMC3::Pythia8ToHepMC3::fill_next_event(Pythi
            a8::Pythia&, HepMC3::GenEvent*, int)':
  >> 235    /tmp/vavolkl/spack-stage/spack-stage-dual-readout-0.0.2-fgymttq7gpq
            wpvxjqlgdjvplqotqdxw5/spack-src/Gen/include/Pythia8ToHepMC3.h:38:60
            : error: invalid conversion from 'const Pythia8::Info*' to 'Pythia8
            ::Info*' [-fpermissive]
     236             return fill_next_event( pythia.event, evt, ievnum, &pythia
            .info, &pythia.settings);
     237    
```        

Not sure if it is worth fixing - as soon as I finish with the transition to edm4hep I'll try to merge this code with the existing pythia interface in the framework.

**Comment by SanghyunKo** [comment]
Hi @vvolkl, this is my bad. Actually this part is bare copy & paste of the original HepMC source code ([HepMC3/interfaces/pythia8/include/Pythia8/Pythia8ToHepMC3.h](https://gitlab.cern.ch/hepmc/HepMC3/-/blob/master/interfaces/pythia8/include/Pythia8/Pythia8ToHepMC3.h)). According to HepMC repository, the original one is already updated to work with Pythia version > 8.301.

I don't remember why I copied this header file in my repository at the moment; probably I can make it to use the original HepMC3 header. But indeed, if we transit to EDM4hep, then we may not need to get bothered by it.

**Comment by vvolkl** [comment]
Sure, as you prefer -- I'll just have to remove this repo from the automated builds in the meantime, as we need pythia8303 for something else, but I don't think that's too much of a problem right now.

**Comment by SanghyunKo** [comment]
Hi @vvolkl, after almost a year(!), I think this issue can now finally be closed as all simulation workflows are now migrated to key4hep (event generation relies on k4Gen).

I've added git action to test compilation on `/cvmfs/sw.hsf.org/key4hep/setup.sh`, so it should be compatible with other key4hep stacks - except that the digitization process relies on the external dependency [SimSiPM](https://github.com/EdoPro98/SimSiPM) (the current git action will install it temporarily in advance of the actual test)

I created a separate issue #17 to track it. I'll communicate with the authors and follow up on the issue there.
